.github/workflows/preview-deployments.yml

```diff
@@ -0,0 +1,128 @@
+# CI to release the project for Linux, Windows, and MacOS
+
+name: Upload Preview Packages
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
Is this needed ? If we do PRs each time the action will run in the PR and again when merged to main, do we want this ?

Reply:
@AntoineRR , I have done it for ease of use. I would love to click on the final merge and get the artifacts.

Reply:
Ok makes sense ^^